src/Minitwit.Infrastructure/Repository/AuthorRepository.cs

```diff
@@ -81,37 +83,51 @@ public ICollection<Cheep> GetCheepsByAuthor(Guid id, int page)
 
         int pageSizeIndex = (page - 1) * PageSize;
 
-        if (author.Cheeps.Count < pageSizeIndex + PageSize)
-            return author.Cheeps.ToList<Cheep>().GetRange(pageSizeIndex, author.Cheeps.Count - pageSizeIndex)
+        if (cheeps.Count < pageSizeIndex + PageSize)
+            return cheeps.ToList().GetRange(pageSizeIndex, cheeps.Count - pageSizeIndex)
                 .OrderByDescending(c => c.TimeStamp).ToList();
-        if(author.Cheeps.Count > PageSize) return author.Cheeps.ToList<Cheep>().GetRange(pageSizeIndex,PageSize).OrderByDescending(c => c.TimeStamp).ToList();
-        return author.Cheeps.OrderByDescending(c => c.TimeStamp).ToList();
+        if(cheeps.Count > PageSize) return cheeps.ToList().GetRange(pageSizeIndex,PageSize).OrderByDescending(c => c.TimeStamp).ToList();
+        return cheeps.OrderByDescending(c => c.TimeStamp).ToList();
+    }
+
+    public ICollection<Cheep> GetCheepsByAuthorAndFollowing(Guid id)
+    {
+        ICollection<Cheep> cheeps = new List<Cheep>();
+        cheeps = cheeps.Concat(GetCheepsByAuthor(id)).ToList();
```

Comment:
We have to concat it, yes. Since the Add() method only takes singular entries, we cannot simply add all elements from a collection. The ToList() is also necessary since an ICollection is returned, at we cannot implicitly work with it as an interface